### ISSUE TITLE
Possible to move Ulauncher's window?

### ISSUE THREAD

**Issue by Cantello** [Ulauncher/Ulauncher#741]
Simple as that: is it possible to move ulauncher's window someplace else? Currently it is always shown in the middle slightly above the center of my primary display. I'd like to move it to the top of the display but cannot move the window (neither via mouse, Alt+mouse or context menu) and there are no prefs that allow setting of the window placement.
Am I missing something? (using Ulauncher 5.11.0 with Gnome 3.38 on Pop!_OS 20.10).

**Comment by friday** [comment]
Sorry for the lack of response.

Yeah, there is no option for this currently. It might be a good thing to be able to change, but adding settings for everything that someone wants would make the code base more complicated and the settings view hard to navigate.

It's a reasonable suggestion though, so it might happen. Will keep the issue open.

**Comment by troycurtisjr** [comment]
Is this a duplicate of #718  ?

**Comment by friday** [comment]
I closed the other one instead. It was older, but this one is explained better imo.

**Comment by orenmizr** [comment]
@friday but why not put the box in the middle of the screen ? or maybe i have a bug.... it is located 25% from the top it seems

**Comment by friday** [comment]
@orenmizr How would you center it? It's not fixed height. It grows and shrinks. Would you want it to move up and down?

**Comment by orenmizr** [comment]
i came from pop os that are using it centered. don't remember it was ever high. Audi from window that used launchy. 

when results is truncated by fixed number you can make that assumption. the whole idea is narrow as you type with fuzzy search. a lot of results is not good.

please look at pop os in VM and tell my what you think. they also managed to bake in a window switcher. i installed all of then and non of them worked. (4 from extension page). this has do much potential to be the goto launcher for Linux

**Comment by RedBearAK** [comment]
@friday 

Since I come from both Spotlight (macOS) and then Albert Launcher (on various Linux DEs), I too would very much like to see Ulauncher be able to be moved away from the center of the screen. It's often quite handy to be able to see something on the screen while looking up a file or doing calculations in Spotlight/Albert up in the corner. 

Once upon a time, Spotlight was attached to the top-right corner of the screen. It would just drop down from the corner when you clicked on the magnifying glass icon. Then Apple made a detached version several years ago, and it was in the center of the screen and couldn't be moved. I believe Apple quickly learned from their users that not everyone was happy with this arrangement. Since then subsequent versions of macOS would have the Spotlight search dialog still default to being centered, but the dialog was able to be grabbed and moved back to the corner (or wherever you wanted it). And it would stay there forever. 

Albert just has a simple checkbox in its settings to "always center" the Albert search dialog. If you uncheck the box you can move Albert to the corner and it will remember the new location and pop up there whenever it is called. The Albert window is always draggable, but it just won't remember the location until you uncheck the "center window" box in the settings. 

In other words, it's fine to leave the default as popping up in the center of the screen. A lot of users are perfectly happy with that. But some of us find a launcher far less useful if it doesn't allow itself to be moved out of the way so we can see something we might need to reference in the middle of the screen. And a single checkbox for "always center search dialog" that can be disabled by the user should be all that is necessary to support this. Just allow the window to be draggable and save the window location relative to the nearest screen corner if the user moves the window. (I say "relative to the nearest screen corner" so that it sort of "stays in place" if the screen resolution changes, which is what the user will normally want.)

**Comment by friday** [comment]
@RedBearAK You're free to send a PR to the v6 branch if you can make it possible to drag the window, and if you want to add a setting for it to remember and relaunch at the previous offset that's even better. I just don't want any hacky window offset preference, and this issue has very low prio for me, because there's a mountain of other things.

**Comment by friday** [comment]
Fixed in the PR linked above for the V6 branch. V6 won't be released for a while, but we made a tracking issue if you want to be notified #869

The way it works (or "will work") is that the top part of the Ulauncher window around the text input is draggable to move the window now, and most of the Preferences window is too, excluding the scrollbar and the hotkey input field.

**Comment by RedBearAK** [comment]
@friday 

> The way it works (or "will work") is that the top part of the Ulauncher window around the text input is draggable to move the window now, and most of the Preferences window is too, excluding the scrollbar and the hotkey input field.

That sounds great. Looking forward to the eventual v6 release.
